atlasdb-ete-tests/src/test/java/com/palantir/atlasdb/ete/CassandraMultinodeEteTest.java

```diff
@@ -0,0 +1,228 @@
+import com.palantir.docker.compose.connection.Container;
+import com.palantir.docker.compose.connection.DockerPort;
+import com.palantir.docker.compose.connection.waiting.HealthCheck;
+import com.palantir.docker.compose.connection.waiting.SuccessOrFailure;
+import com.palantir.docker.compose.execution.DockerComposeExecArgument;
+import com.palantir.docker.compose.execution.DockerComposeExecOption;
+import com.palantir.giraffe.command.Command;
+import com.palantir.giraffe.command.CommandResult;
+import com.palantir.giraffe.command.Commands;
+import com.palantir.giraffe.host.HostAccessors;
+import com.palantir.giraffe.host.HostControlSystem;
+
+public class CassandraMultinodeEteTest {
+
+    private static final Gradle GRADLE_PREPARE_TASK =
+            Gradle.ensureTaskHasRun(":atlasdb-ete-test-utils:buildCassandraImage");
+
+    private static final int TIMELOCK_SERVER_PORT = 3828;
+
+    private static final String CONTAINER_LOGS_DIRECTORY = "container-logs/cassandra-multinode";
+
+    private static final List<String> CASSANDRA_NODES = ImmutableList.of("cassandra1", "cassandra2", "cassandra3");
+
+    private static final long MAX_CASSANDRA_NODE_DOWN_MILLIS = 30000;
+    private static final long MAX_CASSANDRA_NODES_RUNNING_MILLIS = 3000;
+
+    private static final DockerComposeRule MULTINODE_CASSANDRA_SETUP = DockerComposeRule.builder()
+            .file("docker-compose.multiple-cassandra.yml")
+            .waitingForService("cassandra1", Container::areAllPortsOpen)
+            .waitingForService("cassandra2", Container::areAllPortsOpen)
+            .waitingForService("cassandra3", Container::areAllPortsOpen)
+            .waitingForService("ete1", toBeReady())
+            .saveLogsTo(CONTAINER_LOGS_DIRECTORY)
+            .build();
+
+
+    @ClassRule
+    public static final RuleChain PREPARED_DOCKER_SETUP = RuleChain
+            .outerRule(GRADLE_PREPARE_TASK)
+            .around(MULTINODE_CASSANDRA_SETUP);
+
+    @Test
+    public void shouldRunTransactionsWithAllCassandraNodesRunningWithoutUnacceptableDelay()
+            throws InterruptedException, IOException {
+        TodoResource clientToSingleNode = createClientFor(TodoResource.class, asPort("ete1"));
+
+        long transactionStartTime = System.currentTimeMillis();
+        assertAddTodoTransactionWasSuccessful(clientToSingleNode);
+        long transactionEndTime = System.currentTimeMillis();
+
+        long transactionTimeWithAllNodesRunning = transactionEndTime - transactionStartTime;
+
+        assertThat("transactionTimeWithAllNodesRunning",
+                transactionTimeWithAllNodesRunning,
+                is(lessThan(MAX_CASSANDRA_NODES_RUNNING_MILLIS)));
+
+        String container = CASSANDRA_NODES.get(0);
+        checkNodetoolStatus(MULTINODE_CASSANDRA_SETUP.containers().container(container), "UN", 3);
+    }
+
+    @Test
+    public void shouldRunTransactionsAfterCassandraNodeIsShutDownWithoutUnacceptableDelay()
```

Comment:
Nit: Not a fan of the double-negative "WithoutUnacceptable". Consider: `shouldRunTransactionFastEnoughWithOneCassandraNodeDown`